bootstrap/frontend/src/main/scala/io/udash/bootstrap/datepicker/UdashDatePicker.scala

```diff
@@ -0,0 +1,373 @@
+
+import scala.scalajs.js
+import scala.util.Try
+
+/** Wrapper for <a href="http://eonasdan.github.io/bootstrap-datetimepicker/">Bootstrap 3 Datepicker</a>. */
+class UdashDatePicker private[datepicker](val date: Property[ju.Date],
+                                          val options: ReadableProperty[UdashDatePicker.DatePickerOptions],
+                                          override val componentId: ComponentId)
+  extends UdashBootstrapComponent with Listenable[UdashDatePicker, UdashDatePicker.DatePickerEvent] with StrictLogging {
+  import UdashDatePicker._
+
+  import scalatags.JsDom.all._
+
+  private lazy val dateString: Property[String] = date.transform(dateToMomentString, momentStringToDate)
+  private lazy val inp = input(id := componentId.id, tpe := "text", BootstrapStyles.Form.formControl).render
+  private lazy val jQInput = jQ(inp).asDatePicker()
+
+  /** Shows date picker widget. */
+  def show(): Unit =
+    jQInput.dpData().show()
+
+  /** Hides date picker widget. */
+  def hide(): Unit =
+    jQInput.dpData().hide()
+
+  /** Toggle date picker widget visibility. */
+  def toggle(): Unit =
+    jQInput.dpData().toggle()
+
+  /** Enables date input. */
+  def enable(): Unit =
+    jQInput.dpData().enable()
+
+  /** Disables date input. */
+  def disable(): Unit =
+    jQInput.dpData().disable()
+
+  lazy val render: dom.Element = {
+    jQInput.datetimepicker(optionsToJsDict(options.get))
+
+    dateString.listen(s => jQInput.value(s))
+    options.listen(opts => jQInput.dpData().options(optionsToJsDict(opts)))
+
+    jQInput.dpData().date(dateString.get)
+
+    jQInput.on("dp.change", (_: dom.Element, ev: JQueryEvent) => {
+      val date: MomentFormatWrapper = ev.asInstanceOf[DatePickerChangeJQEvent].date
+      val oldDate: MomentFormatWrapper = ev.asInstanceOf[DatePickerChangeJQEvent].oldDate
+      dateString.set(Try(date.format(options.get.format)).getOrElse(null))
+      fire(UdashDatePicker.DatePickerEvent.Change(this, momentToDate(date), momentToDate(oldDate)))
+    })
+    jQInput.on("dp.hide", (_: dom.Element, ev: JQueryEvent) => {
+      val date: MomentFormatWrapper = ev.asInstanceOf[DatePickerHideJQEvent].date
+      fire(UdashDatePicker.DatePickerEvent.Hide(this, momentToDate(date)))
+    })
+    jQInput.on("dp.show", (_: dom.Element, ev: JQueryEvent) =>
+      fire(UdashDatePicker.DatePickerEvent.Show(this))
+    )
+    jQInput.on("dp.error", (_: dom.Element, ev: JQueryEvent) => {
+      val date: MomentFormatWrapper = ev.asInstanceOf[DatePickerErrorJQEvent].date
+      fire(UdashDatePicker.DatePickerEvent.Error(this, momentToDate(date)))
+    })
+
+    inp
+  }
+
+  private def optionsToJsDict(options: UdashDatePicker.DatePickerOptions): js.Dictionary[js.Any] = {
+    import scalajs.js.JSConverters._
+    val dict = js.Dictionary[js.Any]("format" -> options.format)
+    options.locale.foreach(dict.update("locale", _))
+    dict.update("dayViewHeaderFormat", options.dayViewHeaderFormat)
```

Comment:
``js.Dictionary`` has a vararg ``apply`` - you could use it instead of multiple updates